pkg/api/osa/vlabs/types.go

```diff
@@ -0,0 +1,99 @@
+package vlabs
+
+// OpenShiftCluster complies with the ARM model of resource definition in a JSON
+// template.
+type OpenShiftCluster struct {
+	ID         string                `json:"id,omitempty"`
+	Location   string                `json:"location,omitempty"`
+	Name       string                `json:"name,omitempty"`
+	Plan       *ResourcePurchasePlan `json:"plan,omitempty"`
+	Tags       map[string]string     `json:"tags,omitempty"`
+	Type       string                `json:"type,omitempty"`
+	Properties Properties            `json:"properties,omitempty"`
+}
+
+// ResourcePurchasePlan defines the resource plan as required by ARM for billing
+// purposes.
+type ResourcePurchasePlan struct {
+	Name          string `json:"name,omitempty"`
+	Product       string `json:"product,omitempty"`
+	PromotionCode string `json:"promotionCode,omitempty"`
+	Publisher     string `json:"publisher,omitempty"`
+}
+
+// Properties represents the cluster definition.
+type Properties struct {
+	ProvisioningState ProvisioningState `json:"provisioningState,omitempty"`
+	OpenShiftVersion  string            `json:"openShiftVersion,omitempty"`
+	// TODO: need to clarify external API for PublicHostname and decide what to
+	// implement when.  Allow users to specify a PublicHostname then have them
```

Comment:
Asking users to precreate CNAME will also need validation before cluster create is declared a  success and is also error prone as it needs manual steps+there is always potential for a user to misread documentation. OC should create an FQDN for the customer and return it to them.